src/Input/DownshiftSelect/Select.tsx

```diff
@@ -0,0 +1,355 @@
+    }),
+
+    ...(!isUndefined(selectedItemExternal) && {
+      selectedItem: selectedItemExternal,
+    }),
+    ...(isFunction(onSelectedItemChangeExternal) && {
+      onSelectedItemChange: ({ selectedItem }) =>
+        onSelectedItemChangeExternal(selectedItem),
+    }),
+
+    ...downshift,
+  });
+
+  const [helperId] = useId(1, 'description-');
+
+  const handleClear = () => {
+    if (isFunction(onClear)) {
+      onClear();
+    }
+    reset();
+    setInputValue('');
+    // reset does not call onSelectedItemChange on reset, but we want this
+    // for convenience
+    if (isFunction(onSelectedItemChangeExternal)) {
+      onSelectedItemChangeExternal(null);
+    }
+  };
+
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const onFocusInternal = () => {
+    setIsFocused(true);
+  };
+  const onBlurInternal = () => {
+    setIsFocused(false);
+  };
+  const onFocus = () => {
+    if (isFunction(onFocusExternal)) {
+      onFocusExternal();
+    }
+    onFocusInternal();
+  };
+  const onBlur = () => {
+    if (isFunction(onBlurExternal)) {
+      onBlurExternal();
+    }
+    onBlurInternal();
+  };
+
+  return (
+    <Container data-testid="container">
+      {label && (
+        <Label
+          {...getLabelProps()}
+          data-active={isFocused}
+          data-disabled={disabled}
+          data-invalid={invalid}
+          data-testid="label"
+        >
+          {label}
+        </Label>
+      )}
+      <Combobox
+        {...getComboboxProps()}
+        data-disabled={disabled}
+        data-invalid={invalid}
+        data-active={isFocused}
+        onClick={disabled ? noop : openMenu}
+        data-testid="combobox"
+      >
+        <Input
+          {...getInputProps({
+            disabled: disabled,
+            readOnly: disableTyping,
+            placeholder: placeholder,
+            onFocus,
+            onBlur,
+            ...props,
+          })}
+          aria-invalid={invalid}
+          {...(helperText && { 'aria-describedby': helperId })}
+          {...(selectedItem && { title: selectedItem.label })}
+          data-testid="input"
+        />
+        <IndicatorsContainer data-testid="indicators-container">
+          {isLoading && <LoadingIndicator data-testid="loading-indicator" />}
+          {selectedItem && (
+            <ClearButton
+              type="button"
+              onClick={handleClear}
+              aria-label="clear selection"
+              data-testid="clear-button"
+            >
+              <CloseCircleSolidIcon />
+            </ClearButton>
+          )}
+          <ToggleButton
+            type="button"
+            {...getToggleButtonProps({
+              disabled,
+              onClick: e => e.stopPropagation(),
+            })}
+            aria-label="toggle menu"
+            data-testid="toggle-button"
+            // getToggleButtonProps adds a ref here, but functional components don't support them
+            ref={undefined}
```

Comment:
Sorry, I forgot what will happen if we remove this line?

Reply:
It just gives a warning that functional components don't support refs. Updated the comments.